uwblueprint/headsupguys
# SWR Data Fetching

## Notion ticket link

[Fix Data Fetching to use SWR](https://www.notion.so/uwblueprintexecs/Fix-Data-Fetching-to-use-SWR-cec0f10ff9824e3f8239fa4393c0181d)

## Implementation description

- Migrated from `useEffect` to `useSWR` to include loading placeholder during fetching
- Refactored modules and tools page to consist of static components (header) and dynamic components (containing all data)

## Steps to test

1. Run branch locally
2. Visit modules and tools page - ensure static components (header) appear upon render and spinner shows to indicate loading / error message
3. Visit builder page for a module - ensure all data is loaded after the loading screen (doesn't say "Untitled Module" before fetching is complete)

## What should reviewers focus on?

- Code organization

## Checklist
